bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+            }
+            initPolling(COMMAND_REFRESH_SECONDS);
+        } catch (InterruptedException e) {
+            logger.debug("command interupted", e);
+        } catch (IOException e) {
+            logger.debug("Exception executing command", e);
+            initPolling(COMMAND_REFRESH_SECONDS);
+        } catch (NotAuthorizedException e) {
+            logger.debug("Authorization Exception sending command", e);
+            configure();
+        }
+    }
+
+    /**
+     * Configures this thing
+     */
+    private void configure() {
+        clearPolling();
+        firstRun = true;
+
+        IAqualinkConfiguration configuration = getConfig().as(IAqualinkConfiguration.class);
+        String username = configuration.userName;
+        String password = configuration.password;
+        String confSerialId = configuration.serialId;
+        String confApiKey = configuration.apiKey;
+
+        if (StringUtils.isBlank(username)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "username must not be empty");
+            return;
+        }
+
+        if (StringUtils.isBlank(password)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "password must not be empty");
+            return;
+        }
+
+        if (StringUtils.isNotBlank(confApiKey)) {
+            this.apiKey = confApiKey;
+        } else {
+            this.apiKey = DEFUALT_API_KEY;
+        }
+
+        this.refresh = configuration.refresh.intValue() > MIN_REFRESH_SECONDS ? configuration.refresh.intValue()
+                : MIN_REFRESH_SECONDS;
+
+        try {
+            AccountInfo accountInfo = client.login(username, password, apiKey);
+            sessionId = accountInfo.getSessionId();
+            if (sessionId == null) {
+                throw new IOException("Response from controller not valid");
+            }
+            logger.debug("SessionID {}", sessionId);
+
+            Device[] devices = client.getDevices(apiKey, accountInfo.getAuthenticationToken(), accountInfo.getId());
+
+            if (devices.length == 0) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No registered devices found");
+                return;
+            }
+
+            if (StringUtils.isNotBlank(confSerialId)) {
+                serialNumber = confSerialId.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();
+                if (!Arrays.stream(devices).anyMatch(device -> device.getSerialNumber().equals(serialNumber))) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "No Device for given serialId found");
+                    return;
+                }
+            } else {
+                serialNumber = devices[0].getSerialNumber();
+            }
+
+            initPolling(0);
+
+        } catch (InterruptedException | IOException e) {
```

Comment:
If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/